fastapi/dependencies/utils.py

```diff
@@ -245,7 +245,13 @@ def is_scalar_sequence_field(field: ModelField) -> bool:
 
 def get_typed_signature(call: Callable[..., Any]) -> inspect.Signature:
     signature = inspect.signature(call)
-    globalns = getattr(call, "__globals__", {})
+    nsobj = call
+    while hasattr(nsobj, "__wrapped__"):
+        # The __wrapped__ attribute is set by decorators, e.g. functools.wraps.
+        # This while loop allows rereferencing forward references on decorated
+        # methods.
+        nsobj = nsobj.__wrapped__  # type: ignore
```

Comment:
It can be replaced with `inspect.unwrap`:
```suggestion
    nsobj = inspect.unwrap(call)
```

Reply:
Looks like that was added in python 3.4 and the earliest supported version of python for FastAPI is 3.6, so sure. It also checks for cycles (which `typing.get_type_hints` does not).